mayastor/tests/rebuild.rs

```diff
@@ -0,0 +1,1051 @@
+            None,
+            None,
+        )
+        .await;
+        children.push(Child {
+            hdl: test.grpc_handle(name).await.unwrap(),
+            share_uri,
+        });
+    }
+
+    // Create a nexus with a single healthy child.
+    let nexus_hdl = &mut test.grpc_handle("ms1").await.unwrap();
+    create_nexus(nexus_hdl, vec![children[1].share_uri.clone()]).await;
+
+    let degraded_children = children[2 ..= 4].to_vec();
+    assert_eq!(degraded_children.len(), 3);
+
+    // Add children and pause rebuilds.
+    for child in &degraded_children {
+        add_child(nexus_hdl, &child.share_uri, true).await;
+        pause_rebuild(nexus_hdl, &child.share_uri).await;
+    }
+    assert_eq!(
+        get_num_rebuilds(nexus_hdl).await as usize,
+        degraded_children.len()
+    );
+
+    // Resume rebuilds and wait for completion then remove the children.
+    for child in &degraded_children {
+        resume_rebuild(nexus_hdl, &child.share_uri)
+            .await
+            .expect("Failed to resume rebuild");
+        assert!(
+            wait_for_rebuild_completion(
+                nexus_hdl,
+                &child.share_uri,
+                Duration::from_secs(10),
+            )
+            .await
+        );
+        remove_child(nexus_hdl, &child.share_uri).await;
+    }
+    assert_eq!(get_num_rebuilds(nexus_hdl).await, 0);
+
+    // Add the children back again
+    for child in &degraded_children {
+        add_child(nexus_hdl, &child.share_uri, true).await;
+    }
+
+    // Wait for rebuilds to complete
+    for child in &degraded_children {
+        assert!(
+            wait_for_rebuild_completion(
+                nexus_hdl,
+                &child.share_uri,
+                Duration::from_secs(10),
+            )
+            .await
+        );
+    }
+}
+
+/// Test rebuild while running front-end I/O.
+/// Note: This test can take some time to complete because it is running fio and
+/// then comparing the contents of the children to make sure they are in-sync.
+#[tokio::test]
+async fn rebuild_with_load() {
+    init_tracing();
+    let test = start_infrastructure("rebuild_with_load").await;
+    let nexus_hdl = &mut test.grpc_handle("ms1").await.unwrap();
+    let ms2 = &mut test.grpc_handle("ms2").await.unwrap();
+    let ms3 = &mut test.grpc_handle("ms3").await.unwrap();
+
+    const CHILD_SIZE_MB: u64 = 100;
+
+    // Create a nexus with 1 child.
+    let child1 =
+        bdev_create_and_share(ms2, Some(CHILD_SIZE_MB), Some("disk1".into()))
+            .await;
+    create_nexus(nexus_hdl, vec![child1.clone()]).await;
+
+    // Connect to nexus over NVMf.
+    let nexus_uri = publish_nexus(nexus_hdl).await;
+    let nexus_tgt = nvmf_connect(nexus_uri.clone());
+
+    // Run fio against nexus.
+    let (s, r) = unbounded::<i32>();
+    let nvmf_tgt = nexus_tgt.clone();
+    std::thread::spawn(move || {
+        s.send(common::fio_verify_size(&nvmf_tgt, NEXUS_SIZE))
+            .unwrap();
```

Comment:
probably don't want to unwrap here and it can block the test, should unwrap in the test thread.

Reply:
Thanks - I've added an error message instead.